libdevice/include/sanitizer_device_utils.hpp

```diff
@@ -0,0 +1,31 @@
+//==-- sanitizer_device_utils.hpp - Declaration for sanitizer global var ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "spir_global_var.hpp"
+#include <cstdint>
+
+// Treat this header as system one to workaround frontend's restriction
+#pragma clang system_header
+
+#ifdef __SPIR__
+#ifdef __SYCL_DEVICE_ONLY__
+
+// declaration
+extern SPIR_GLOBAL_VAR __SYCL_GLOBAL__ uintptr_t __AsanShadowMemoryGlobalStart;
+extern SPIR_GLOBAL_VAR __SYCL_GLOBAL__ uintptr_t __AsanShadowMemoryGlobalEnd;
+extern SPIR_GLOBAL_VAR __SYCL_GLOBAL__ uintptr_t __AsanShadowMemoryLocalStart;
+extern SPIR_GLOBAL_VAR __SYCL_GLOBAL__ uintptr_t __AsanShadowMemoryLocalEnd;
+extern SPIR_GLOBAL_VAR __SYCL_GLOBAL__ uintptr_t __AsanShadowMemoryPrivateStart;
```

Comment:
Looks like __AsanShadowMemoryPrivateStart/__AsanShadowMemoryPrivateEnd are not used. Maybe we'd better to remove them first?

Reply:
Done